"Plugin did not respond" error during applying opendistro_role

Hello! Thank you for your provider, it is really convenient and helpful!

Unfortunately I faced multiple issues with it, the most frustrating is "Plugin did not respond".

```
Terraform v1.1.4
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.74.1
+ provider registry.terraform.io/hashicorp/http v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/random v3.0.1
+ provider registry.terraform.io/phillbaker/elasticsearch v2.0.0-beta.4
```

Here is my tf-code example:
```
provider "elasticsearch" {
  url                 = "https://search-{{ ES DNS NAME }}.eu-west-1.es.amazonaws.com"
  aws_assume_role_arn = module.es.es_admin_role_arn
  healthcheck         = false
}

...

resource "elasticsearch_opendistro_role" "this" {
  role_name   = var.role_name
  description = var.issue

  cluster_permissions = ["cluster_composite_ops"]

  index_permissions {
    index_patterns  = var.index_patterns
    allowed_actions = ["unlimited"]
  }
}

resource "elasticsearch_opendistro_user" "this" {
  username    = var.user_name
  password    = local.user_password

  lifecycle {
    ignore_changes = [password]
  }
}
```

And here is the error example(I'm getting multiple errors with different roles/users):
```
╷
│ Error: Plugin did not respond
│ 
│   with module.opendistro_groups["{{ GROUP NAME }}"].elasticsearch_opendistro_user.this,
│   on ../modules/opendistro/main.tf line 29, in resource "elasticsearch_opendistro_user" "this":
│   29: resource "elasticsearch_opendistro_user" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-elasticsearch_v2.0.0-beta.4 plugin:

panic: set item just set doesn't exist

goroutine 225 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setSet(0xc000cb2858, 0xc000b5fe10, 0x1, 0x1, 0x1d28640, 0xc000cb28a0, 0xc000561e00, 0x1f627dd, 0xc000b8a3c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/field_writer_map.go:328 +0xc65
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc000cb2858, 0xc000b5fe10, 0x1, 0x1, 0x1d28640, 0xc000cb28a0, 0x1, 0x1)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/field_writer_map.go:107 +0x250
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc000cb2858, 0xc000b5fe10, 0x1, 0x1, 0x1d28640, 0xc000cb28a0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/field_writer_map.go:89 +0x3f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc0005c9c80, 0x1f627dd, 0x12, 0x1d28640, 0xc000cb28a0, 0x1, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/resource_data.go:227 +0x1df
github.com/phillbaker/terraform-provider-elasticsearch/es.resourceElasticsearchOpenDistroRoleRead(0xc0005c9c80, 0x1d01580, 0xc0005be780, 0x28e9c20, 0xc000400480)
        github.com/phillbaker/terraform-provider-elasticsearch/es/resource_elasticsearch_opendistro_role.go:158 +0x58a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0001935c0, 0x21614a8, 0xc0002f96c0, 0xc0005c9c80, 0x1d01580, 0xc0005be780, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/resource.go:290 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0001935c0, 0x21614a8, 0xc0002f96c0, 0xc000388d90, 0x1d01580, 0xc0005be780, 0xc000514598, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/resource.go:564 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00000c510, 0x21614a8, 0xc0002f96c0, 0xc0002f9700, 0xc0002f96c0, 0x100b665, 0x1e700a0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc000331980, 0x2161550, 0xc0002f96c0, 0xc00007ff80, 0xc000331980, 0xc000291740, 0xc000256ba0)
        github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/server/server.go:297 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x1ebf100, 0xc000331980, 0x2161550, 0xc000291740, 0xc00007ff20, 0x0, 0x2161550, 0xc000291740, 0xc000ab5560, 0x120)
        github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001ff880, 0x216cfd8, 0xc000402900, 0xc00032d700, 0xc00069b170, 0x28abbd0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.33.2/server.go:1210 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0001ff880, 0x216cfd8, 0xc000402900, 0xc00032d700, 0x0)
        google.golang.org/grpc@v1.33.2/server.go:1533 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000386d0, 0xc0001ff880, 0x216cfd8, 0xc000402900, 0xc00032d700)
        google.golang.org/grpc@v1.33.2/server.go:871 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.33.2/server.go:869 +0x1fd

Error: The terraform-provider-elasticsearch_v2.0.0-beta.4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

A couple of things from debug log: https://gist.github.com/sergkondr/c88848f7219ac7b72e0bf62854bed97c

meanwhile adding some logs output with TF_LOG=info:
```
2022-02-28T00:28:32.034+0200 [INFO]  backend/local: apply calling Apply
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "path.module"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_index_template.index_template#destroy"
2022-02-28T00:28:32.040+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_opendistro_ism_policy.ism_policy#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_opendistro_roles_mapping.master_user_arn#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_opendistro_roles_mapping.master_user_arn#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "path.module"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_index_template.index_template#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_opendistro_ism_policy.ism_policy#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "path.module"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "path.module"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "aws_route53_record.opensearch#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "elasticsearch_opendistro_roles_mapping.master_user_arn#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.key"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "each.value"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "aws_route53_record.opensearch#destroy"
2022-02-28T00:28:32.041+0200 [INFO]  ReferenceTransformer: reference not found: "path.module"
2022-02-28T00:28:32.049+0200 [INFO]  provider: configuring client automatic mTLS
2022-02-28T00:28:32.124+0200 [INFO]  provider.terraform-provider-aws_v4.2.0_x5: configuring server automatic mTLS: timestamp=2022-02-28T00:28:32.123+0200
2022-02-28T00:28:32.332+0200 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/hashicorp/aws\"]" changed the config value, but that value is unused
2022-02-28T00:28:34.069+0200 [INFO]  provider: configuring client automatic mTLS
2022-02-28T00:28:34.100+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: configuring server automatic mTLS: timestamp=2022-02-28T00:28:34.099+0200
2022-02-28T00:28:34.168+0200 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/phillbaker/elasticsearch\"]" changed the config value, but that value is unused
2022-02-28T00:28:34.169+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [WARN] Truncating attribute path of 0 diagnostics for TypeSet: timestamp=2022-02-28T00:28:34.169+0200
2022-02-28T00:28:34.169+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [WARN] Truncating attribute path of 0 diagnostics for TypeSet: timestamp=2022-02-28T00:28:34.169+0200
2022-02-28T00:28:34.169+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [WARN] Truncating attribute path of 0 diagnostics for TypeSet: timestamp=2022-02-28T00:28:34.169+0200
2022-02-28T00:28:34.169+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [WARN] Truncating attribute path of 0 diagnostics for TypeSet: timestamp=2022-02-28T00:28:34.169+0200
2022-02-28T00:28:34.169+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [WARN] Truncating attribute path of 0 diagnostics for TypeSet: timestamp=2022-02-28T00:28:34.169+0200
2022-02-28T00:28:34.169+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [WARN] Truncating attribute path of 0 diagnostics for TypeSet: timestamp=2022-02-28T00:28:34.169+0200
2022-02-28T00:28:34.170+0200 [WARN]  Provider "registry.terraform.io/phillbaker/elasticsearch" produced an invalid plan for module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["security_manager"], but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .users: planned value cty.NullVal(cty.Set(cty.String)) does not match config value cty.SetValEmpty(cty.String)
      - .hosts: planned value cty.NullVal(cty.Set(cty.String)) does not match config value cty.SetValEmpty(cty.String)
2022-02-28T00:28:34.170+0200 [WARN]  Provider "registry.terraform.io/phillbaker/elasticsearch" produced an invalid plan for module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["all_access"], but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .hosts: planned value cty.NullVal(cty.Set(cty.String)) does not match config value cty.SetValEmpty(cty.String)
      - .users: planned value cty.NullVal(cty.Set(cty.String)) does not match config value cty.SetValEmpty(cty.String)
module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["security_manager"]: Creating...
module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["all_access"]: Creating...
2022-02-28T00:28:34.170+0200 [INFO]  Starting apply for module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["security_manager"]
2022-02-28T00:28:34.170+0200 [INFO]  Starting apply for module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["all_access"]
2022-02-28T00:28:34.171+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [INFO] Using AWS: us-east-1: timestamp=2022-02-28T00:28:34.171+0200
2022-02-28T00:28:34.171+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 [INFO] Using AWS: us-east-1: timestamp=2022-02-28T00:28:34.171+0200
2022-02-28T00:28:34.172+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0-beta.4: 2022/02/28 00:28:34 EmptyStaticCreds: static credentials are empty: timestamp=2022-02-28T00:28:34.172+0200
2022-02-28T00:28:34.174+0200 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = transport is closing"
2022-02-28T00:28:34.174+0200 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = transport is closing"
2022-02-28T00:28:34.195+0200 [ERROR] vertex "module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn[\"all_access\"]" error: Plugin did not respond
2022-02-28T00:28:34.216+0200 [ERROR] vertex "module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn[\"security_manager\"]" error: Plugin did not respond
╷
│ Warning: Deprecated Resource
│ 
│   with module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["security_manager"],
│   on ../opensearch/role_mapping.tf line 16, in resource "elasticsearch_opendistro_roles_mapping" "master_user_arn":
│   16: resource "elasticsearch_opendistro_roles_mapping" "master_user_arn" {
│ 
│ elasticsearch_opendistro_roles_mapping is deprecated, please use opensearch_roles_mapping resource instead.
│ 
│ (and one more similar warning elsewhere)
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["security_manager"],
│   on ../opensearch/role_mapping.tf line 16, in resource "elasticsearch_opendistro_roles_mapping" "master_user_arn":
│   16: resource "elasticsearch_opendistro_roles_mapping" "master_user_arn" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.opensearch.elasticsearch_opendistro_roles_mapping.master_user_arn["all_access"],
│   on ../opensearch/role_mapping.tf line 16, in resource "elasticsearch_opendistro_roles_mapping" "master_user_arn":
│   16: resource "elasticsearch_opendistro_roles_mapping" "master_user_arn" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
```

> @mahmoud-hafez-aws my OpenSearch cluster version is 1.1

 In terraform INFO logs in the noticed the following,  the provider wasn't able to find the credentials chain!!

````
-----------------------------------------------------]: timestamp=2022-04-20T14:04:07.698+0200
elasticsearch_opensearch_roles_mapping.mapper: Still creating... [40s elapsed]
2022-04-20T14:04:17.708+0200 [INFO]  provider.terraform-provider-elasticsearch_v2.0.0: 2022/04/20 14:04:17 NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors: timestamp=2022-04-20T14:04:17.707+0200
2022-04-20T14:04:17.709+0200 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Canceled desc = context canceled"
2022-04-20T14:04:17.717+0200 [ERROR] vertex "elasticsearch_opensearch_roles_mapping.mapper" error: Request cancelled
````

I use the assume role and tried to place it in both levels in provider "aws" and in provider "elasticsearch" and the role is already part of the "security_manages" role in the cluster but unfortunately, still, it didn't work.

--- a/es/provider.go
+++ b/es/provider.go
@@ -320,10 +320,18 @@ func getClient(conf *ProviderConf) (interface{}, error) {
 
 	if m := awsUrlRegexp.FindStringSubmatch(conf.parsedUrl.Hostname()); m != nil && conf.signAWSRequests {
 		log.Printf("[INFO] Using AWS: %+v", m[1])
-		opts = append(opts, elastic7.SetHttpClient(awsHttpClient(m[1], conf, map[string]string{})), elastic7.SetSniff(false))
+		client, err := awsHttpClient(m[1], conf, map[string]string{})
+		if err != nil {
+			return nil, err
+		}
+		opts = append(opts, elastic7.SetHttpClient(client), elastic7.SetSniff(false))
 	} else if awsRegion := conf.awsRegion; conf.awsRegion != "" && conf.signAWSRequests {
 		log.Printf("[INFO] Using AWS: %+v", awsRegion)
-		opts = append(opts, elastic7.SetHttpClient(awsHttpClient(awsRegion, conf, map[string]string{})), elastic7.SetSniff(false))
+		client, err := awsHttpClient(awsRegion, conf, map[string]string{})
+		if err != nil {
+			return nil, err
+		}
+		opts = append(opts, elastic7.SetHttpClient(client), elastic7.SetSniff(false))
 	} else if conf.insecure || conf.cacertFile != "" {
 		opts = append(opts, elastic7.SetHttpClient(tlsHttpClient(conf, map[string]string{})), elastic7.SetSniff(false))
 	} else if conf.token != "" {
@@ -423,10 +431,18 @@ func getClient(conf *ProviderConf) (interface{}, error) {
 
 		if m := awsUrlRegexp.FindStringSubmatch(conf.parsedUrl.Hostname()); m != nil && conf.signAWSRequests {
 			log.Printf("[INFO] Using AWS: %+v", m[1])
-			opts = append(opts, elastic6.SetHttpClient(awsHttpClient(m[1], conf, map[string]string{})), elastic6.SetSniff(false))
+			client, err := awsHttpClient(m[1], conf, map[string]string{})
+			if err != nil {
+				return nil, err
+			}
+			opts = append(opts, elastic6.SetHttpClient(client), elastic6.SetSniff(false))
 		} else if awsRegion := conf.awsRegion; conf.awsRegion != "" && conf.signAWSRequests {
 			log.Printf("[INFO] Using AWS: %+v", conf.awsRegion)
-			opts = append(opts, elastic6.SetHttpClient(awsHttpClient(awsRegion, conf, map[string]string{})), elastic6.SetSniff(false))
+			client, err := awsHttpClient(awsRegion, conf, map[string]string{})
+			if err != nil {
+				return nil, err
+			}
+			opts = append(opts, elastic6.SetHttpClient(client), elastic6.SetSniff(false))
 		} else if conf.insecure || conf.cacertFile != "" {
 			opts = append(opts, elastic6.SetHttpClient(tlsHttpClient(conf, map[string]string{})), elastic6.SetSniff(false))
 		} else if conf.token != "" {
@@ -501,10 +517,18 @@ func getKibanaClient(conf *ProviderConf) (interface{}, error) {
 
 		if m := awsUrlRegexp.FindStringSubmatch(conf.parsedUrl.Hostname()); m != nil && conf.signAWSRequests {
 			log.Printf("[INFO] Using AWS: %+v", m[1])
-			opts = append(opts, elastic7.SetHttpClient(awsHttpClient(m[1], conf, headers)), elastic7.SetSniff(false))
+			client, err := awsHttpClient(m[1], conf, headers)
+			if err != nil {
+				return nil, err
+			}
+			opts = append(opts, elastic7.SetHttpClient(client), elastic7.SetSniff(false))
 		} else if awsRegion := conf.awsRegion; conf.awsRegion != "" && conf.signAWSRequests {
 			log.Printf("[INFO] Using AWS: %+v", awsRegion)
-			opts = append(opts, elastic7.SetHttpClient(awsHttpClient(awsRegion, conf, headers)), elastic7.SetSniff(false))
+			client, err := awsHttpClient(awsRegion, conf, headers)
+			if err != nil {
+				return nil, err
+			}
+			opts = append(opts, elastic7.SetHttpClient(client), elastic7.SetSniff(false))
 		} else if conf.insecure || conf.cacertFile != "" {
 			opts = append(opts, elastic7.SetHttpClient(tlsHttpClient(conf, headers)))
 		} else if conf.token != "" {
@@ -589,18 +613,18 @@ func awsSession(region string, conf *ProviderConf) *awssession.Session {
 	return awssession.Must(awssession.NewSessionWithOptions(sessOpts))
 }
 
-func awsHttpClient(region string, conf *ProviderConf, headers map[string]string) *http.Client {
+func awsHttpClient(region string, conf *ProviderConf, headers map[string]string) (*http.Client, error) {
 	session := awsSession(region, conf)
 	// Call Get() to ensure concurrency safe retrieval of credentials. Since the
 	// client is created in many go routines, this synchronizes it.
 	_, err := session.Config.Credentials.Get()
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 	signer := awssigv4.NewSigner(session.Config.Credentials)
 	client, err := aws_signing_client.New(signer, session.Config.HTTPClient, conf.awsSig4Service, region)
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 
 	rt := WithHeader(client.Transport)
@@ -610,7 +634,7 @@ func awsHttpClient(region string, conf *ProviderConf, headers map[string]string)
 	}
 	client.Transport = rt
 
-	return client
+	return client, nil
 }
 
 func tokenHttpClient(conf *ProviderConf, headers map[string]string) *http.Client {

